packages/nuka/src/slider-list.tsx

```diff
@@ -0,0 +1,231 @@
+    // x-translation values we need to limit to.
+    const clampIndices: number[] | null = useMemo(() => {
+      if (disableEdgeSwiping && !wrapAround) {
+        const dotIndexes = getDotIndexes(
+          slideCount,
+          slidesToScroll,
+          scrollMode,
+          slidesToShow,
+          wrapAround,
+          cellAlign
+        );
+        return [dotIndexes[0], dotIndexes[dotIndexes.length - 1]];
+      }
+      return null;
+    }, [
+      cellAlign,
+      disableEdgeSwiping,
+      scrollMode,
+      slideCount,
+      slidesToScroll,
+      slidesToShow,
+      wrapAround,
+    ]);
+
+    const initialValue = useMemo(() => {
+      // When wrapAround is enabled, we show the slides 3 times
+      const totalCount = wrapAround ? 3 * slideCount : slideCount;
+      const slideSize = 100 / totalCount;
+      let initialValue = wrapAround ? -slideCount * slideSize : 0;
+
+      if (cellAlign === Alignment.Right && slidesToShow > 1) {
+        const excessSlides = slidesToShow - 1;
+        initialValue += slideSize * excessSlides;
+      }
+
+      if (cellAlign === Alignment.Center && slidesToShow > 1) {
+        const excessSlides = slidesToShow - 1;
+        // Half of excess is on left and half is on right when centered
+        const excessLeftSlides = excessSlides / 2;
+        initialValue += slideSize * excessLeftSlides;
+      }
+
+      return initialValue;
+    }, [cellAlign, slideCount, slidesToShow, wrapAround]);
+
+    const positioning = useMemo(() => {
+      const percentOffsetForSlideProps = {
+        cellAlign,
+        currentPosition,
+        initialValue,
+        slideCount,
+        transition: !isAnimating || slideAnimation === 'fade' ? 1 : transition,
+        wrapAround,
+      };
+
+      const slideBasedOffset = getPercentOffsetForSlide({
+        ...percentOffsetForSlideProps,
+        currentSlide,
+      });
+
+      // Special-case this. It's better to return undefined rather than a
+      // transform of 0 pixels since transforms can cause flickering in chrome.
+      if (draggedOffset === 0 && slideBasedOffset === 0) {
+        return undefined;
+      }
+
+      let clampOffsets: number[] | null = null;
+      if (clampIndices) {
+        clampOffsets = clampIndices.map((index) =>
+          getPercentOffsetForSlide({
+            ...percentOffsetForSlideProps,
+            currentSlide: index,
+          })
+        );
+      }
+
+      const clampedDraggedOffset = clampOffsets
+        ? // Offsets are seemingly backwards because the rightmost slide creates
+          // the most negative translate value
+          `clamp(${clampOffsets[1]}%, ${draggedOffset}px, ${clampOffsets[0]}%)`
+        : `${draggedOffset}px`;
+
+      return `translate3d(${
+        draggedOffset ? clampedDraggedOffset : `${slideBasedOffset}%`
+      }, 0, 0)`;
+
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [transition === 1 ? draggedOffset : transition]);
```

Comment:
Is it dangerous to have a dynamic dependency list here?

Reply:
It was essential for preventing duplicate positioning and still allow the initial drag event to fire.